src/Scripting/CSharp/Hosting/ObjectFormatter/CSharpTypeNameFormatter.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.Scripting.Hosting;
+
+namespace Microsoft.CodeAnalysis.CSharp.Scripting.Hosting
+{
+    public class CSharpTypeNameFormatter : CommonTypeNameFormatter
+    {
+        protected override CommonPrimitiveFormatter PrimitiveFormatter { get; }
```

Comment:
Yes, it's intended to be a part of the public API.  From our experience building similar functionality in the EE, we know that there's a lot of interest in tweaking the existing behavior, rather than replacing it entirely.